ethclient_utils/client.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (C) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package ethclient_utils
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ava-labs/subnet-evm/rpc"
+)
+
+var ErrInvalidEndpoint = errors.New("invalid rpc endpoint")
+
+func DialWithConfig(ctx context.Context, endpoint string, httpHeaders, queryParams map[string]string) (ethclient.Client, error) {
```

Comment:
We can just pass in a `config.APIConfig` object here. That would be better aligned with the naming of the function, as well as encapsulate everything we need in a single type derived directly from the `Config` object.

Reply:
I created a local struct. I cannot use the struct from the config package otherwise there is import cycle error.

Reply:
Are you sure? It looks cleaner with your initial proposition.
Go build takes care about the two structs consistency. It fails if the fields are not aligned.

Reply:
Yeah, I'm sure. Besides duplicating type definitions to satisfy the dependency graph, by specifying the struct in the parameter list (and requiring the caller to construct an instance of that struct) it is not immediately clear from the callsite that the struct does not have any side effects or additional functionality. By passing the parameters directly, it is clear that there are no such side effects, even though the callsite requires more parameters.

The best solution would be to incorporate `DialWithConfig` into `config.APIConfig` directly, so that we can use just that one type throughout the application. This refactor is outside the scope of this PR and issue though, so my preference is to defer that to a later change.